### PR TITLE
Update Assistant API

### PR DESCRIPTION
**Pull request by alperozturk96**

### CODE REVIEW

File: library/src/main/java/com/owncloud/android/lib/resources/assistant/GetTaskListRemoteOperation.kt [hunk 1]
```diff
@@ -18,14 +18,14 @@ import com.owncloud.android.lib.resources.OCSRemoteOperation
 import com.owncloud.android.lib.resources.assistant.model.TaskList
 import org.apache.commons.httpclient.HttpStatus
 
-class GetTaskListRemoteOperation(private val appId: String) : OCSRemoteOperation<TaskList>() {
+class GetTaskListRemoteOperation(private val taskType: String) : OCSRemoteOperation<TaskList>() {
```
Comment (tobiasKaminsky) [inline library/src/main/java/com/owncloud/android/lib/resources/assistant/GetTaskListRemoteOperation.kt:21]: If we have a data class TaskType, we can use it here, right?

File: sample_client/build.gradle [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 buildscript {
     dependencies {
-        classpath 'com.android.tools.build:gradle:8.7.2'
+        classpath 'com.android.tools.build:gradle:8.7.3'
```
Comment (tobiasKaminsky) [inline sample_client/build.gradle:14]: Also here. Please split it into another pr.

File: library/build.gradle [hunk 3]
```diff
@@ -18,7 +18,7 @@ buildscript {
         junit_version = '4.13.2'
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:8.7.2'
+        classpath 'com.android.tools.build:gradle:8.7.3'
```
[the remark this answers is not in the document]
Comment (alperozturk96) [reply library/build.gradle:21]: I can't use with Files client. Client and library must have the same version, otherwise it will not compile.

File: sample_client/build.gradle (hunk 2 above)
Comment (alperozturk96) [reply sample_client/build.gradle:14]: Same here. I have to test the client app, otherwise it's not compiling.

File: library/build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply library/build.gradle:21]: Indeed, when you are using settings.gradle to link both projects.
If you use git commit hash and build it via jitpack.io, this is not needed :) 

But splitting later is fine :+1:

[the remark this answers is not in the document]
Comment (alperozturk96) [reply library/build.gradle:21]: https://github.com/nextcloud/android-library/pull/1599/commits/fab1e27e64a8636a09bd6ce5ac36c41e01965905

File: library/src/main/java/com/owncloud/android/lib/resources/assistant/GetTaskListRemoteOperation.kt (hunk 1 above)
Comment (alperozturk96) [reply library/src/main/java/com/owncloud/android/lib/resources/assistant/GetTaskListRemoteOperation.kt:21]: https://github.com/nextcloud/android-library/pull/1599/commits/fab1e27e64a8636a09bd6ce5ac36c41e01965905

**Comment by tobiasKaminsky** [comment]
CI test fails due to compilation:
```
> Task :library:compileDebugAndroidTestKotlin FAILED
e: file:///drone/src/library/src/androidTest/java/com/owncloud/android/lib/resources/assistant/AssistantIT.kt:29:43 Unresolved reference 'types'.
e: file:///drone/src/library/src/androidTest/java/com/owncloud/android/lib/resources/assistant/AssistantIT.kt:42:53 Argument type mismatch: actual type is 'kotlin.String', but 'com.owncloud.android.lib.resources.assistant.model.TaskTypeData' was expected.
```

File: library/src/androidTest/java/com/owncloud/android/lib/resources/assistant/AssistantIT.kt [hunk 4]
```diff
@@ -18,30 +21,53 @@ import org.junit.Test
 class AssistantIT : AbstractIT() {
     @Before
     fun before() {
-        testOnlyOnServer(NextcloudVersion.nextcloud_28)
+        testOnlyOnServer(NextcloudVersion.nextcloud_30)
```
Comment (tobiasKaminsky) [inline library/src/androidTest/java/com/owncloud/android/lib/resources/assistant/AssistantIT.kt:24]: Why the change?
Does this mean that it will not work with NC28?

Comment (alperozturk96) [reply library/src/androidTest/java/com/owncloud/android/lib/resources/assistant/AssistantIT.kt:24]: The Text Processing API is deprecated and replaced by the Task Processing API since Nextcloud 30.

https://github.com/nextcloud/android/issues/13476

**Review by tobiasKaminsky — APPROVED** [review]
